scripts/gha/build_desktop.py

```diff
@@ -0,0 +1,145 @@
+    script_absolute_path = utils.get_vcpkg_installation_script_path()
+    # Example: ./external/vcpkg/bootstrap-sh
+    utils.run_command([script_absolute_path])
+
+  # for each desktop platform, there exists a vcpkg response file in the repo
+  # (external/vcpkg_<triplet>_response_file.txt) defined for each target triplet
+  vcpkg_triplet = utils.get_vcpkg_triplet(arch)
+  vcpkg_response_file_path = os.path.join(os.getcwd(), 'external', 
+                      'vcpkg_' + vcpkg_triplet + '_response_file.txt')
+
+  # Eg: ./external/vcpkg/vcpkg install @external/vcpkg_x64-osx_response_file.txt 
+  # --disable-metrics
+  utils.run_command([vcpkg_executable_file_path, 'install',
+                     '@' + vcpkg_response_file_path, '--disable-metrics'])
+
+
+def cmake_configure(build_dir, arch, build_tests=True, config=None):
+  """ CMake configure.
+
+  If you are seeing problems when running this multiple times,
+  make sure to clean/delete previous build directory.
+
+  Args:
+   build_dir (str): Output build directory.
+   arch (str): Platform Architecture (example: 'x64').
+   build_tests (bool): Build cpp unit tests.
+   config (str): Release/Debug config.
+          If its not specified, cmake's default is used (most likely Debug).
+  """   
+  cmd = ['cmake', '-S', '.', '-B', build_dir]
+  
+  # If generator is not specifed, default for platform is used by cmake, else 
+  # use the specified value
+  if config:
+    cmd.append('-DCMAKE_BUILD_TYPE={0}'.format(config))
+  if build_tests:
+    cmd.append('-DFIREBASE_CPP_BUILD_TESTS=ON')
+    cmd.append('-DFIREBASE_FORCE_FAKE_SECURE_STORAGE=ON')
+
+  vcpkg_toolchain_file_path = os.path.join(os.getcwd(), 'external',
+                                           'vcpkg', 'scripts', 
+                                           'buildsystems', 'vcpkg.cmake')
+  cmd.append('-DCMAKE_TOOLCHAIN_FILE={0}'.format(vcpkg_toolchain_file_path))
+
+  vcpkg_triplet = utils.get_vcpkg_triplet(arch)
+  cmd.append('-DVCPKG_TARGET_TRIPLET={0}'.format(vcpkg_triplet))
+  
+  # TODO: Remove this once firestore is included in the build and everything works
+  cmd.append('-DFIREBASE_INCLUDE_FIRESTORE=OFF')
+  utils.run_command(cmd)
+
+
+def main():
+  args = parse_cmdline_args()
+  
+  # Ensure that the submodules are initialized and updated
+  # Example: vcpkg is a submodule (external/vcpkg)
+  utils.run_command(['git', 'submodule', 'init'])
+  utils.run_command(['git', 'submodule', 'update'])
+
+  # Install platform dependent cpp dependencies with vcpkg 
+  install_cpp_dependencies_with_vcpkg(args.arch)
+
+  # CMake configure
+  cmake_configure(args.build_dir, args.arch, args.build_tests, args.config)
+
+  # CMake build 
+  # cmake --build build -j 8 
+  cmd = ['cmake', '--build', args.build_dir, '-j', str(os.cpu_count())] 
+  if args.target:
+    # Example:  cmake --build build -j 8 --target firebase_app firebase_auth
+    cmd.append('--target')
+    cmd.extend(args.target)
+  utils.run_command(cmd)
```

Comment:
not cmd = ['cmake', '--build', args.build_dir, '-j', str(os.cpu_count())] + ['--target', args.target] if args.target else []?

Maybe I'm biased as a former Perl programmer ;)

Reply:
So we will have to put parantheses around the second part and that was making it harder to read. 
```
l = ['a', 'b'] + ['c', 'd'] if something else []
# something = True
# l = ['a', 'b', 'c', 'd']
# something = False
# l = []

l = ['a', 'b'] + (['c', 'd'] if something else [])
```